catalog/spatial/registry/registry-admin-modules/registry-admin-local-ui/src/main/webapp/js/model/Association.js

```diff
@@ -0,0 +1,250 @@
+                    return Association.SegmentIds;
+                },
+                relatedModel: function () {
+                    return Association.SegmentId;
+                }
+            },
+            {
+                type: Backbone.One,
+                key: 'topSegment',
+                relatedModel: function () {
+                    return Segment.Segment;
+                }
+            }
+        ],
+        defaults: {
+            associations: [],
+            associationSegments: [],
+            topSegment: null
+        },
+        populateFromModel: function (associations) {
+            var model = this;
+            var collectionModel = this.get('associations');
+            this.dataModel = associations;
+            _.each(associations, function (association) {
+                var newAssociation = new Association.Association();
+                newAssociation.populateFromModel(association, model.get('topSegment'));
+                collectionModel.add(newAssociation);
+            });
+            this.populateAssociationSegmentIds(this.get('topSegment'));
+        },
+        addAssociation: function (source, target, type) {
+            var association = new Association.Association({
+                id: generateId(),
+                sourceId: source,
+                targetId: target,
+                type: type
+            });
+            association.populateFromModel(undefined, this.get('topSegment'));
+            this.get('associations').add(association);
+            return association;
+        },
+        removeAssociation: function (id) {
+            var removedAssociation = _.find(this.get('associations').models, function (association) {
+                return association.get('id') === id;
+            });
+            this.get('associations').remove(removedAssociation);
+            return removedAssociation;
+        },
+        removeSegment: function (segment) {
+            var model = this;
+            var updatedAssociations = [];
+            var associations = this.get('associations').models;
+            var segId = segment.get('segmentId');
+            _.each(associations, function (association) {
+                if (association.get('sourceId') !== segId && association.get('targetId') !== segId) {
+                    updatedAssociations.push(association);
+                }
+            });
+            this.set('associations', updatedAssociations);
+
+            var seg = _.find(this.get('associationSegments').models, function (seg) {
+                return seg.get('segmentId') === segId;
+            });
+            this.get('associationSegments').remove(seg);
+            _.each(segment.get('segments').models, function(curSeg){
+               if(FieldDescriptors.isCustomizableSegment(curSeg.get('segmentType'))){
+                   model.removeSegment(curSeg);
+               }
+            });
+            wreqr.vent.trigger('associationSegmentRemoved',segId);
```

Comment:
Instead of doing a wreqr event, the views could listen to the segment collection for remove events.  This comments applies to other uses of wreqr in this model.